### PR TITLE
[SYCL][NFC] Fix static code analysis concerns

### PR DESCRIPTION
**Pull request by mlychkov**
Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

File: clang/lib/Sema/SemaSYCL.cpp [hunk 1]
```diff
@@ -254,7 +254,7 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
       }
       // Specifically check if the math library function corresponding to this
       // builtin is supported for SYCL
-      unsigned BuiltinID = (Callee ? Callee->getBuiltinID() : 0);
```
Comment (MrSidims) [inline clang/lib/Sema/SemaSYCL.cpp]: Just curious, what was wrong here?

Comment (mlychkov) [reply clang/lib/Sema/SemaSYCL.cpp]: Static code analyzer somehow worried that Callee is checked on nullptr here however is dereferenced earlier at line 235 without  this check.
But actually it is already tested at line 216 and no additional check is required.

Comment (mlychkov) [reply clang/lib/Sema/SemaSYCL.cpp]: Oh, I missed resetting Callee at line 217.

File: clang/lib/Sema/SemaSYCL.cpp [hunk 2]
```diff
@@ -215,6 +215,8 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
     if (FunctionDecl *Callee = e->getDirectCallee()) {
       Callee = Callee->getCanonicalDecl();
+      assert(Callee && "Device function canonical decl must be available");
```
Comment (AlexeySachkov) [inline clang/lib/Sema/SemaSYCL.cpp]: Why do you need this assertion? `Callee` is already checked for `nullptr` within `if` statement on 216

**Pull request merged**